### PR TITLE
Update exercises and tests for fall 2021

### PR DESCRIPTION
**Pull request by bentondavidl**

### CODE REVIEW

File: pythonBasics3Test.py [hunk 1]
```diff
@@ -1,47 +1,53 @@
...
+    def test_starts_with_non_number(self):
 
-        self.assertEqual(pythonBasics3.ends_with_consonant("Once upon a dark, cold night"), True)
+        self.assertEqual(pythonBasics3.starts_with_non_number("Once upon a time, I was 18"), True)
 
-        self.assertEqual(pythonBasics3.ends_with_consonant("What did Steve just say"), True)
+        self.assertEqual(pythonBasics3.starts_with_non_number("5 weekdays each week"), False)
 
-        self.assertEqual(pythonBasics3.ends_with_consonant("I once swam with sharks"), True)
+        self.assertEqual(pythonBasics3.starts_with_non_number("-5 is what I got on my quiz"), True)
```
Comment (beauseate) [inline pythonBasics3Test.py:14]: Why is -5 a non number?

File: pythonBasics3.py [hunk 2]
```diff
@@ -7,33 +7,36 @@
...
-# # Part A. starts_with_number
-# Define a function ends_with_number(s) that takes a string and returns true
-# if it ends with a number and false otherwise.
+# # Part A. starts_with_non_number
+# Define a function starts_with_non_number(s) that takes a string and returns true
+# if it starts with anything that isn't a number and false otherwise.
 # (For our purposes, a number is any character that is 0,1,2,3,4,5,6,7,8, or 9.)
 # Note: Be sure to use RegEx!
-def ends_with_number(s):
+def starts_with_non_number(s):
```
Comment (beauseate) [inline pythonBasics3.py:15]: Maybe name func `does_not_start_with_number`

File: pythonBasics3.py [hunk 3]
```diff
@@ -7,33 +7,36 @@
...
-# if it ends with a consonant and false otherwise.
-# (For our purposes, a consonant is any letter other than A, E, I, O, U.)
-# Note: Be sure to use RegEx and it works for both upper and lower case and for nonletters!
-def ends_with_consonant(s):
+# # Part B. multiple_words
+# Define a function multiple_words(s) that takes a string and returns true
+# if it contains multiple words and false otherwise.
+# (For our purposes, a string has multiple words if it contains any whitespace characters.)
+# Note: Be sure to use RegEx and it works for any whitespace character 
+# and there's at least one character on either side!
```
Comment (beauseate) [inline pythonBasics3.py]: "with at least one..."

File: pythonBasics3Test.py [hunk 4]
```diff
@@ -1,47 +1,53 @@
...
+        self.assertEqual(pythonBasics3.multiple_words("That's 10/10"), True)
 
-        self.assertEqual(pythonBasics3.ends_with_number("It was not a cat"), False)
+        self.assertEqual(pythonBasics3.multiple_words(" "), False)
 
-        self.assertEqual(pythonBasics3.ends_with_number("I once swam with sharks"), False)
+        self.assertEqual(pythonBasics3.multiple_words("Different\twhitespace"), False)
 
-        self.assertEqual(pythonBasics3.ends_with_number("Blue plus purple equals 5"), True)
+        self.assertEqual(pythonBasics3.multiple_words("It's-all-one-word"), False)
```
Comment (beauseate) [inline pythonBasics3Test.py:31]: One more edge case:

" a " == False

**Review by beauseate — CHANGES_REQUESTED** [review]

File: pythonBasics3Test.py (hunk 1 above)
Comment (bentondavidl) [reply pythonBasics3Test.py:14]: The goal of this is to be able to use character classes. The first character (`-`) is not part of the digit character class (`\d`). Therefore, the first character is not a number in this case.

File: pythonBasics3.py (hunk 2 above)
Comment (bentondavidl) [reply pythonBasics3.py:15]: I thought about using that but I didn't want to make it too long. It's not great as it is, but I didn't want it longer. It also hints at the solution of using character classes

Comment (beauseate) [reply pythonBasics3.py:15]: This'll solve both, `starts_without_positive_num` or something to that extent

Comment (bentondavidl) [reply pythonBasics3.py:15]: Unless Dr. Najjar wants me to change it, I will just make it clear in the instructions that the first character must be a number

Comment (nanajjar) [reply pythonBasics3.py:15]: I'd go with starts_with_non_number. It is clear that way what the function needs to do.

**Pull request merged**